src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py

```diff
@@ -394,6 +420,7 @@ def classpath_for_context(context):
     # all targets to finish.
     with self.invalidated(relevant_targets,
                           invalidate_dependents=True,
+                          invalidate_direct_checker=JvmCompile.enable_strict_deps,
```

Comment:
That method is global, but it is used as the _default_ value of the option, if a target doesn't specify it directly. Each target may have a different value for this field, as specified in https://github.com/pantsbuild/pants/blob/master/src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py#L829